src/components/TimingToast/TimingToast.tsx

```diff
@@ -2,21 +2,22 @@ import { Notification } from '@components/Notification'
 import { TOGGLE_TIMING_TOAST } from '@constants/eventTypes'
 import { useStyles } from '@helpers/hooks/useStyles'
 import emitter from '@lib/emitter'
-import PropTypes from 'prop-types'
-import React, { useCallback, useEffect, useState } from 'react'
+import { FC, useCallback, useEffect, useState } from 'react'
 
 import styles from './TimingToast.module.scss'
 
-/**
- * TimingToast - stateless presentational component
- * @param {object} props - props
- * @param {string|array} props.className - list of class names
- * @return {object} An object of children element
- */
-const TimingToast = ({ className }) => {
+type Type = 'info' | 'success' | 'warning' | 'alert'
+
+interface TimingToastProps {
+  className?: string | string[]
+}
+
+export const TimingToast: FC<TimingToastProps> = ({ className }) => {
   const [isActive, setActive] = useState(false)
   const [message, setMessage] = useState('')
-  const [type, setType] = useState('info')
+  const [type, setType] = useState<Type>('info')
+
+  let autoHideTimer: NodeJS.Timeout | null = null
```

Comment:
zastanawiam się czy nie lepszy byłby useRef dla jakich edge caseów, taka zmienna w przypadku stanu i odświeżenia nie będzie znowu nullem ? :P

Reply:
nie, nie działa

Reply:
testowałem to

Reply:
Spoczko to daje approvke